src/Passwordless/PasswordlessClient.cs

```diff
@@ -33,34 +54,28 @@ public PasswordlessClient(PasswordlessOptions options)
     {
     }
 
-    private static readonly string SdkVersion =
-        typeof(PasswordlessClient).Assembly.GetName().Version?.ToString(3) ??
-        // This should never happen, unless the assembly had its metadata trimmed
-        "unknown";
-
-    private readonly HttpClient _http = new(new PasswordlessHttpHandler(http, disposeClient), true)
+    // We validate the options right before making a request, so that we don't
+    // throw an exception during the service registration.
+    private void ValidateOptions()
     {
-        BaseAddress = new Uri(options.ApiUrl),
-        DefaultRequestHeaders =
+        if (string.IsNullOrWhiteSpace(options.ApiSecret))
         {
-            {
-                "ApiSecret",
-                options.ApiSecret
-            },
-            {
-                "Client-Version",
-                $".NET-{SdkVersion}"
-            }
+            throw new InvalidOperationException(
+                "Missing Passwordless API Secret. " +
+                "Please make sure a valid value is configured."
+            );
         }
-    };
+    }
 
     /// <inheritdoc />
     public async Task<RegisterTokenResponse> CreateRegisterTokenAsync(
-        RegisterOptions options,
+        RegisterOptions registerOptions,
         CancellationToken cancellationToken = default)
     {
+        ValidateOptions();
```

Comment:
Having to include this in every method is suboptimal, but simple. An alternative would be to put this code inside `PasswordlessHttpHandler`. Open for suggestions.

Reply:
Simple is good, good is simple ;)